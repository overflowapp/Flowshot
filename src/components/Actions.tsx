import * as React from 'react';

const Actions = ({ children }) => <div className="pb5 ph6 flex flex-noshrink mtauto">{children}</div>;

export default Actions;
