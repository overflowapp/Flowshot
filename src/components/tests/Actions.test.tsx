import * as React from 'react';
import { shallow } from 'enzyme';
import Actions from '../Actions';

describe('Actions', function() {
    const props = {
        children: <a className="button">Begin new session</a>,
    };
    const MountedActions = shallow(<Actions {...props} />);

    it('should render button correctly', function() {
        expect(MountedActions.find('a').exists()).toBe(true);
        expect(MountedActions.find('a').hasClass('button')).toBe(true);
        expect(MountedActions.text()).toBe('Begin new session');
    });

    it('pass snapshot', function() {
        expect(MountedActions).toMatchSnapshot();
    });
});
