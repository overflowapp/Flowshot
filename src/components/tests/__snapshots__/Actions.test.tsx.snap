// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions pass snapshot 1`] = `
<div
  className="pb5 ph6 flex flex-noshrink mtauto"
>
  <a
    className="button"
  >
    Begin new session
  </a>
</div>
`;
